proxy/envoy/config_test.go

```diff
@@ -241,20 +270,8 @@ func testConfig(r *model.IstioRegistry, instance, envoyConfig, testCase string,
 	if err != nil {
 		t.Fatalf(err.Error())
 	}
-	data, err := ioutil.ReadFile(envoyConfig)
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
 
-	expected, err := ioutil.ReadFile(envoyConfig + ".golden")
-	if err != nil {
-		t.Fatalf(err.Error())
-	}
-
-	// TODO: use difflib to obtain detailed diff
-	if string(expected) != string(data) {
-		t.Errorf("Envoy config %q differs from master copy for %q", envoyConfig, testCase)
-	}
+	compareJSON(envoyConfig, t)
```

Comment:
It might be better to return the error (and diff) and handle logging/t.Errorf in the caller, e.g.

```golang
if diff, eq := compareJSON(envoyConfig); !eq {
    t.Errorf("Failed validating golden artifact: %v", diff)
}
```